Pull in latest changes

alexankitty opened the pull request:
This is the staged PR to bring it in line with latest tachiyomi in an effort to fix extensions not working.
A lot of work will need to be done to bridge the parity between the 2000 commits it's behind.

Rattlehead15 commented:
Hey, thanks for trying to get this fixed! In the meantime, you can use https://github.com/getraid/tachiyomiJ2K-OCR this other fork someone made a couple months ago. Sources work fine there

arkon committed:
Use unified storage location for automatic backups

arkon committed:
Add unified storage location setting

Currently only using it as a replacement for the downloads location.

arkon committed:
Ensure page indicator texts are centered

Maybe fixes #9976

arkon committed:
Require Android 8+

Given that the next stable version of Chrome (120) will require Android 8+, it's
inevitable that the WebView functionality will gradually break. As always, newer
OS versions are recommended for better support with evolving Internet technologies.

According to https://apilevels.com/, Android 8+ still covers 93.7% of Android users.

arkon committed:
Revert "Require Android 8+"

This reverts commit 64c50c1283969dbc16c519f87cad3b5c9001b4cb.

Forgot we need to manage app update checks manually...

arkon committed:
No-op app update checks for Android < 8

This effectively makes it the last release for the older Android versions.

arkon committed:
Show EOL message if update check no-ops due to unsupported Android version

arkon committed:
Minor cleanup

arkon committed:
Restrict line length with ktlint

arkon committed:
Postpone Android 8 requirement

arkon committed:
Lint fixes

ivaniskandar committed:
MangaScreen: Adjust "missing chapter count" item list styling (#10105)

Text style to labelMedium and set secondary alpha to the whole row

arkon committed:
Clean up reading mode / orientation enum classes

Categorizing the reading modes so we can implement a better
selection UI.

arkon committed:
Replace custom preview annotation

AntsyLich committed:
Implement scanlator filter (#8803)

* Implement scanlator filter

* Visual improvement to scanlator filter dialog

* Review changes + Bug fixes

Backup not containing filtered chapters and similar issue fix

* Review Changes + Fix SQL query

* Lint mamma mia

AntsyLich committed:
Trim `Chapter` scanlator value (#10108)

arkon committed:
Add separate default/apply buttons to reading mode/orientation selection dialogs

Related to #3453

arkon committed:
Group mode dialogs together in bottom reader bar

renovate[bot] committed:
Update dependency org.junit.jupiter:junit-jupiter to v5.10.1 (#10109)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update voyager to v1.0.0-rc09 (#10110)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

AntsyLich committed:
ScanlatorFilterDialog: Fix crash when no scanlator (#10111)

AntsyLich committed:
Trim scanlator of chapters in db (#10112)

arkon committed:
Convert create backup dialog to a screen

Allows us more flexibility in adding more options/explanations in the future.

arkon committed:
Update to Compose Compiler 1.5.4 and Kotlin 1.9.20

arkon committed:
Fix EmptyScreen kaomoji looking broken for RTL locales

arkon committed:
Add Compose lint checks

Still need to address most of them though.

arkon committed:
Minor cleanup

arkon committed:
Use immutable collections in presentation-widget module

arkon committed:
Use immutable collections in more places

renovate[bot] committed:
Update voyager to v1.0.0-rc10 (#10127)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

weblate committed:
Translations update from Hosted Weblate (#10102)

Weblate translations




















Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ar/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ca/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/el/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/es/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/es_419/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fil/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/hr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/id/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ja/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ru/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/th/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/tr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/uk/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hans/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hant/
Translation: Tachiyomi/Tachiyomi 0.x

Co-authored-by: Alessandro Jean <14254807+alessandrojean@users.noreply.github.com>
Co-authored-by: Ali Aljishi <ahj696@hotmail.com>
Co-authored-by: Dexroneum <Rozhenkov69@gmail.com>
Co-authored-by: Eduard Ereza Martínez <eduard@ereza.cat>
Co-authored-by: Hiroshi <borlonjhayron1119@gmail.com>
Co-authored-by: InfinityDouki56 <ced.paltep10@gmail.com>
Co-authored-by: La prière <lapriere@users.noreply.hosted.weblate.org>
Co-authored-by: Lzmxya <lzmxya@gmail.com>
Co-authored-by: Milo Ivir <mail@milotype.de>
Co-authored-by: Pitpe11 <giorgos2550@gmail.com>
Co-authored-by: TheKingTermux <achmadmaulana0233@gmail.com>
Co-authored-by: Uzuki Shimamura <hzy980512@126.com>
Co-authored-by: Zero O <godarms2010@live.com>
Co-authored-by: altinat <altinat@duck.com>
Co-authored-by: bapeey <luisrleccar@hotmail.com>
Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: orkan gökçe alaz aşina <examplehuman@outlook.com>
Co-authored-by: winver <kirillstuzhuk@gmail.com>

Ereza committed:
Fix Catalan plurals manually (#10133)

Co-authored-by: arkon <arkon@users.noreply.github.com>

arkon committed:
Bump dependencies

ivaniskandar committed:
Migrate deprecated tooltip components (#10141)

ivaniskandar committed:
Migrate deprecated progress indicator components (#10142)

ivaniskandar committed:
Update Scaffold fork (#10143)

https://android-review.googlesource.com/c/platform/frameworks/support/+/2690433

arkon committed:
Migrate to standard M3 TabIndicator

arkon committed:
Address more Compose lint warnings

ivaniskandar committed:
Remove animateItemPlacement modifier usage (#10146)

weblate committed:
Translations update from Hosted Weblate (#10135)

Weblate translations













Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ca/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/cs/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/da/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/de/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/lv/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ro/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ru/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/sv/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/tr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/uk/
Translation: Tachiyomi/Tachiyomi 0.x

Co-authored-by: Dexroneum <Rozhenkov69@gmail.com>
Co-authored-by: Druvvaldis <druvvaldisr@gmail.com>
Co-authored-by: Eduard Ereza Martínez <eduard@ereza.cat>
Co-authored-by: Lyfja <yassinelaoud@gmail.com>
Co-authored-by: Matyáš Caras <contact@hernikplays.cz>
Co-authored-by: Osyx <ofalkman@gmail.com>
Co-authored-by: Rostyslav Haitkulov <info@ubilling.net.ua>
Co-authored-by: alex <hdhdhfhfbbffhhfhfjfjf@gmail.com>
Co-authored-by: f0roots <f0rootss@gmail.com>
Co-authored-by: orkan gökçe alaz aşina <examplehuman@outlook.com>
Co-authored-by: symegac <97731141+symegac@users.noreply.github.com>

arkon committed:
Migrate to multiplatform string resources (#10147)

* Migrate to multiplatform string resources

* Move plurals translations into separate files

* Fix lint check on generated files

arkon committed:
Replace remaining Android-specific strings

Also renaming the helper composables so it's a bit easier to find/replace everything
in forks.

arkon committed:
Suppress build warnings from MPP modules

ivaniskandar committed:
Migrate XML themes to Compose (#10152)

arkon committed:
Slightly shrink continue reading button size

arkon committed:
Fix searchbar style

Also address some Compose lint warnings.

arkon committed:
Add "Rotate wide pages to fit" for webtoon reader too

Closes #1977

arkon committed:
Use custom threshold for what's consider a low RAM device

arkon committed:
Remove max automatic backups setting

We just always create up to 4 now to simplify it, given the addition of syncing
is going to make this section pretty busy.

AntsyLich committed:
MangaScreen: Fix close in action mode exists from screen (#10160)

* MangaScreen: Fix close in action mode exists from screen

* L

ivaniskandar committed:
Migrate to M3 pull-to-refresh (#10164)

arkon committed:
Exclude some more app state preferences from backups

arkon committed:
Call WheelPicker onSelectionChanged with initial value

Fixes #10157

We realistically only ever use the picker in contexts where we later
confirm or cancel with the selected value, so this is fine. If the caller
wants to ignore the initial value, they can always check if it's distinct
before/after there.

ivaniskandar committed:
Fix PTR extra offset calculation (#10172)

ivaniskandar committed:
Fix PTR initial refreshing state (#10173)

ivaniskandar committed:
Change default PTR colors (#10174)

arkon committed:
Avoid crashing if storage directory can't be read

e.g. when first launching and there's no storage permissions yet.

AntsyLich committed:
Library update notification changes (#10175)

Don't round up notification percentage. Why show 100% when stuff is still updating.

Show same notification when hide notification content is enabled. Just exclude manga titles.

arkon committed:
Bump dependencies

arkon committed:
Minor download location cleanup

arkon committed:
Use unified storage location for local source

arkon committed:
Actually use configured storage location for local source

Fixes #10178

Saud-97 committed:
Anilist decode item description HTML (#10181)

arkon committed:
Add extensions for handling UniFile name/file extensions

arkon committed:
Use UniFile for local source file handling

arkon committed:
Move automatic backups from /backup/automatic to /autobackup

Removes the need to try to create child folders, which simplifies things.

arkon committed:
Show copied to clipboard toast on Samsung devices even if Android 13+

Since OneUI didn't implement the AOSP thing.

arkon committed:
Adjust stats overview icons

Closes #9865
Still sort of weird, but the icons are now always aligned.

arkon committed:
Differ extra attempts to load local series' covers until chapter loading

arkon committed:
Revert "Show copied to clipboard toast on Samsung devices even if Android 13+"

This reverts commit bf524595e27e5a42aceb6df0dae2817cefac63c6.

Apparently it shows a toast, but I don't see it?

arkon committed:
Update UniFile

Which has more correct nullability for some methods and case insensitivity for listFiles where possible.

arkon committed:
Ignore casing when looking for some files/folders

arkon committed:
Remove storage permissions

Requires adjusting some file reading to first copy to a temporary file
in cache that we have permissions to read from. This is only applicable for things
like ZIP files where we need an actual File rather than just some Android content
URI shenanigans.

arkon committed:
Target SDK 30

Need to convert some services into WorkManager jobs before going to 31 and higher.

ivaniskandar committed:
HomeScreen: Add static key for TabNavigator (#10191)

Fixes incorrect tab selection after process death

ivaniskandar committed:
Migrate downloader service to WorkManager (#10190)

arkon committed:
Bump dependencies

arkon committed:
Remove unused resources

arkon committed:
Replace ReaderOrientation icon resources

arkon committed:
Bump dependencies

ivaniskandar committed:
Define storage permission in manifest (#10200)

Mainly for migration process and to avoid breakage on users who doesn't
change their granular storage prefs.

ivaniskandar committed:
PullRefresh: Invoke callback only when initiated by user (#10201)

arkon committed:
Set foreground service types for library update jobs

arkon committed:
Revert overridePendingTransition refactorings

weblate committed:
Translations update from Hosted Weblate (#10148)

Weblate translations



























Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ar/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ca/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/de/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/el/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/es/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fa/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fil/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/hr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/hu/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/id/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ja/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/jv/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/nb_NO/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/pl/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ru/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/sc/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/sv/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/th/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/tr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hant/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/ca/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/es/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/fa/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/hu/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/ja/
Translation: Tachiyomi/Tachiyomi plurals.xml
Translation: Tachiyomi/Tachiyomi strings.xml

Co-authored-by: Ajeje Brazorf <lmelonimamo@yahoo.it>
Co-authored-by: Alessandro Jean <14254807+alessandrojean@users.noreply.github.com>
Co-authored-by: Ali Aljishi <ahj696@hotmail.com>
Co-authored-by: Arash <ara.khoram95@gmail.com>
Co-authored-by: CR0YD <98400750+CR0YD@users.noreply.github.com>
Co-authored-by: Dexroneum <Rozhenkov69@gmail.com>
Co-authored-by: Eduard Ereza Martínez <eduard@ereza.cat>
Co-authored-by: FaCsaba <csab.faz.2@gmail.com>
Co-authored-by: InfinityDouki56 <ced.paltep10@gmail.com>
Co-authored-by: La prière <lapriere@users.noreply.hosted.weblate.org>
Co-authored-by: Lyfja <yassinelaoud@gmail.com>
Co-authored-by: Lzmxya <lzmxya@gmail.com>
Co-authored-by: Milo Ivir <mail@milotype.de>
Co-authored-by: Nepx <anandabaskara@outlook.com>
Co-authored-by: Pitpe11 <giorgos2550@gmail.com>
Co-authored-by: Swyter <swyterzone@gmail.com>
Co-authored-by: TheKingTermux <achmadmaulana0233@gmail.com>
Co-authored-by: Uzuki Shimamura <hzy980512@126.com>
Co-authored-by: Vetle Ledaal <vetle.ledaal@gmail.com>
Co-authored-by: Xavier Giguère (bebewold) <bebewold@gmail.com>
Co-authored-by: aa aa <lpodlewski76@gmail.com>
Co-authored-by: altinat <altinat@duck.com>
Co-authored-by: bittin1ddc447d824349b2 <bittin@reimu.nl>
Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: orkan gökçe alaz aşina <examplehuman@outlook.com>

ivaniskandar committed:
Move workers to foreground service context a bit more safely (#10202)

The system will crash the app if the worker that calls setForeground() finished
before the service runner be able to call Service.startForeground(). This edge
case is not handled by WorkManager and there is no way to check if the required
calls are done.

So here we suspend the worker by an arbitrary duration assuming the transition
to foreground service is done by then.

arkon committed:
Handle content URIs for covers

Co-authored-by: jmir1 <jmir1@users.noreply.github.com>

renovate[bot] committed:
Update dependency com.android.tools.build:gradle to v8.2.0 (#10212)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

arkon committed:
Remove storage permission check when manually creating backups

Co-authored-by: jmir1 <jmir1@users.noreply.github.com>

arkon committed:
Clean up startDownloadNow function a bit

Fixes #9330, I think. If it was even still an issue.

arkon committed:
Force recreate download index cache on upgrade

Fixes #10187

arkon committed:
Migrate ReaderColorFilterView to Compose

arkon committed:
Try to fall back to showing URI for storage location if concrete file path isn't available

Closes #9977

arkon committed:
Minor cleanup

arkon committed:
Fix sharing saved pages from notification

Related to #8327
Deleting doesn't seem to do anything still, but at least doesn't throw an exception.

Also removed behavior of dismissing notification after sharing/deleting pages/backups
in case you want to do something again afterwards. Users can manually dismiss the
notification whenever they want.

arkon committed:
Show average scores in tracker search results

Closes #8280

arkon committed:
Bump dependencies

ivaniskandar committed:
Downloader: Don't queue chapters on GlobalScope (#10217)

This fixes auto-download on library update not working on certain cases.

arkon committed:
Migrate to some newer date/time APIs

arkon committed:
Add basic onboarding screen (#10199)

arkon committed:
More onboarding screen additions

arkon committed:
More onboarding screen additions 2: Electric Boogaloo

weblate committed:
Translations update from Hosted Weblate (#10204)

Weblate translations








Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fa/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fil/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ja/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/lv/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ro/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/ro/
Translation: Tachiyomi/Tachiyomi plurals.xml
Translation: Tachiyomi/Tachiyomi strings.xml

Co-authored-by: Alessandro Jean <14254807+alessandrojean@users.noreply.github.com>
Co-authored-by: Arash <ara.khoram95@gmail.com>
Co-authored-by: Druvvaldis <druvvaldisr@gmail.com>
Co-authored-by: InfinityDouki56 <ced.paltep10@gmail.com>
Co-authored-by: Saft Octavian <saftoctavian@gmail.com>
Co-authored-by: TheKingTermux <achmadmaulana0233@gmail.com>

arkon committed:
Upgrade Voyager

arkon committed:
Minor cleanup/docs

arkon committed:
Fix next local chapter not being indicated as downloaded in transition

arkon committed:
Always show trackers action in MangaScreen

Goes to tracker settings to log in if none are set up.

arkon committed:
Onboarding screen tweaks

- Opposite transition when going back a step
- Don't allow skipping (I don't want to deal with an unset storage location in other places)

arkon committed:
Fix settings screen crashing when saving state

arkon committed:
Remove action to delete saved image in notification

Can just open it and delete from whatever gallery app the user has which has way
more functionality.

Closes #8327

arkon committed:
Avoid starting restore job if already running

We already check in the settings screen where it's triggered, but who knows.
Also addressing some errors for method calls that require SDK 26+ (but don't
fail the build, somehow?).

arkon committed:
Add ability to open available extension websites in WebView

Closes #8628

arkon committed:
Clean up category restoring logic

arkon committed:
Clean up some text alpha modifiers

arkon committed:
Proper check for when to navigate to tracker settings from tracking action

arkon committed:
Clean up manga restoring logic

Some behavior changes:
- It prioritizes new entries, then anything more recently updated
- It copies the more recently updated entry's metadata (description, thumbnail, etc.)

arkon committed:
Remove redundant job setup calls in migrations

We always set them up earlier in the migrations anyway.

arkon committed:
Fix download indexing with changed storage locations

Fixes #10218

arkon committed:
Skip updating unchanged chapters and tracks when restoring backup

arkon committed:
Bump dependencies

arkon committed:
Avoid crashing in SourcePreferencesFragment if source can't be loaded

Should probably wait for sources to definitely be loaded first, but that's
sort of a bigger change and needs to be lifecycle-aware.

ivaniskandar committed:
Refactor onboarding steps

(cherry picked from commit 2ca3ab077192a7e5e2e7a5fb00c303a5a633372e)

arkon committed:
Workaround for broken nav bar icon colors

weblate committed:
Translations update from Hosted Weblate (#10222)

Weblate translations

















Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ar/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/bn/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ca/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/de/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/el/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/es/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fil/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ja/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ko/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/nb_NO/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ru/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/sv/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/tr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hans/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hant/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/bn/
Translation: Tachiyomi/Tachiyomi plurals.xml
Translation: Tachiyomi/Tachiyomi strings.xml

Co-authored-by: Alessandro Jean <14254807+alessandrojean@users.noreply.github.com>
Co-authored-by: Ali-98 <ahj696@hotmail.com>
Co-authored-by: Dexroneum <Rozhenkov69@gmail.com>
Co-authored-by: Eduard Ereza Martínez <eduard@ereza.cat>
Co-authored-by: Hasanur Rahman Biplob <hrbiplob100@gmail.com>
Co-authored-by: InfinityDouki56 <ced.paltep10@gmail.com>
Co-authored-by: Jakob Holkestad Molnes <Jakob.Holkestad.Molnes@gmail.com>
Co-authored-by: Lyfja <yassinelaoud@gmail.com>
Co-authored-by: Lzmxya <lzmxya@gmail.com>
Co-authored-by: Oğuz Ersen <oguz@ersen.moe>
Co-authored-by: Pierre Kim <admin@manateeshome.com>
Co-authored-by: Pitpe11 <giorgos2550@gmail.com>
Co-authored-by: Uzuki Shimamura <hzy980512@126.com>
Co-authored-by: bapeey <luisrleccar@hotmail.com>
Co-authored-by: bittin1ddc447d824349b2 <bittin@reimu.nl>

arkon committed:
Clean up history restoring

arkon committed:
Split restoring logic into smaller classes

arkon committed:
Set foreground service types for remaining jobs

arkon committed:
Target Android 12L (SDK 32)

arkon committed:
Set foreground service type for ExtensionInstallService

ivaniskandar committed:
Target Android 14 (SDK 34) and add permission onboarding step

(cherry picked from commit 9e0068715f3ba3d1627c4b7539b90fb782f8122f)

arkon committed:
Migrate BuildConfig to Gradle Build Files

renovate[bot] committed:
Update dependency com.squareup.okio:okio to v3.7.0 (#10239)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

ivaniskandar committed:
MangaScreenModel: Start downloads in IO dispatcher (#10241)

ivaniskandar committed:
PackageInstallerInstaller: Fix intent used for install session (#10240)

Use explicit intent as it's a requirement when targeting v34+

ivaniskandar committed:
DownloadJob: Network check changes (#10242)

Mostly pulled from WorkManager

weblate committed:
Translations update from Hosted Weblate (#10238)

Weblate translations










Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/es/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ja/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/sq/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/sv/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/tr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/vi/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hant/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/vi/
Translation: Tachiyomi/Tachiyomi plurals.xml
Translation: Tachiyomi/Tachiyomi strings.xml

Co-authored-by: Alessandro Jean <14254807+alessandrojean@users.noreply.github.com>
Co-authored-by: DatTran MLL <tranthanhdat1142003@gmail.com>
Co-authored-by: Lzmxya <lzmxya@gmail.com>
Co-authored-by: Oğuz Ersen <oguz@ersen.moe>
Co-authored-by: Uzuki Shimamura <hzy980512@126.com>
Co-authored-by: bapeey <luisrleccar@hotmail.com>
Co-authored-by: bittin1ddc447d824349b2 <bittin@reimu.nl>
Co-authored-by: lisienskenderi <lisienskenderi@hotmail.com>

arkon committed:
Update ignore paths for translation PRs

arkon committed:
Bump NDK

Just using the same version as J2K for now, we can probably go higher though.

arkon committed:
Bump to latest NDK LTS

renovate[bot] committed:
Update okhttp monorepo to v5.0.0-alpha.12 (#10245)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

arkon committed:
Prevent backing out from initial onboarding

arkon committed:
Use smaller window to calculate fetch interval if there's less total chapters

This is sort of a workaround for sources that tend to only give you the first few and
most recent few chapters, which would have been 28 day intervals before due to
the big gap in the middle.

arkon committed:
Update Compose compiler

arkon committed:
Foundations for partial restores

Related to #3136

arkon committed:
Show MIUI warning more prominently in CreateBackupScreen

arkon committed:
Initial move of restore backup into a separate screen

arkon committed:
Janky workaround for Moko escaped quotes issue

Related: https://github.com/icerockdev/moko-resources/issues/337

arkon committed:
Whoops, accidentally made the create backup button unusable before

arkon committed:
Don't use localized numbers for downloaded image filenames

Probably fixes #10258

arkon committed:
Highlight restore backup setting when navigating from onboarding step

arkon committed:
Don't include settings as defaults when manually creating backup

arkon committed:
Split up BackupCreator into smaller classes

arkon committed:
Move backup models to domain module

arkon committed:
Some domain Track model migrations

arkon committed:
Minor tracking cleanups

arkon committed:
Ensure sufficiently long prefix when creating temp file

Fixes #10265

arkon committed:
Avoid crashing if loading non-read-only private extension

Fixes #10252

arkon committed:
Address some build warnings

arkon committed:
Fix read duration statistic getting inflated when restoring history

arkon committed:
Clean up storage usage info

- Show bar representation of used/total space
- Handle all mounted storages
- Also included a bunch of unrelated immutables changes, sorry

arkon committed:
Tweak page flashing

Closes #10269, maybe. I don't really have something to test with.

arkon committed:
Update Coil

arkon committed:
Simplify storage usage bar UI implementation

arkon committed:
Remove skipped updates notification

Seems to cause more confusion than it's worth.
Will update the UI for the library update skip options to better
explain what they're for later.

arkon committed:
Always show chapter download indicators

- Local chapters are just always "downloaded", but you can't delete them from the app
- Unavailable sources show proper state so long as it's stubbed with the name still, but
  you can't download anything new

arkon committed:
Allow deleting downloaded chapters even if source isn't available

Fixes #9160

arkon committed:
Move backup models back to main app module

I didn't realize the package name change would break compatibility with forks
that still have the old package names...

ivaniskandar committed:
Implement predictive back animation (#10273)

For home screen tabs, Navigator screens and most dialogs

arkon committed:
Hide private installer option for stable builds

arkon committed:
Don't make install permission required during onboarding

Closes #10257

We show a warning banner in the extensions list and also rely on the system
alert popup if someone attempts to install without the permission already
granted.

arkon committed:
Minor exception message cleanup

arkon committed:
Refactor backup option flags to normal data class of booleans

arkon committed:
Add ability to create manual backups with private preferences too

arkon committed:
Minor cleanup/address lint warnings

arkon committed:
Clean up create backup UI

arkon committed:
Filter out empty source preferences when creating backups

arkon committed:
Allow partial restores (library/settings)

Closes #3136

arkon committed:
Shorten restore warning message a bit

arkon committed:
Allow creating backups without library entries

- In case you want a backup of just settings?
- Also disable backup options if dependent option is disabled (and fix being able to toggle disabled items)
- Also fix crash in RestoreBackupScreen due to attempt to parcelize Uri
- Make restore validation message a bit nicer

arkon committed:
Misc refactoring

- Abstract away relative date string building
- Dedupe large update warning logic

arkon committed:
Show next expected update in interval dialog

Related: #9793

arkon committed:
Dedupe common LazyColumn with action at bottom layout

renovate[bot] committed:
Update dependency org.jsoup:jsoup to v1.17.2 (#10277)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

arkon committed:
Don't use reflection for handling backup options as boolean array

Wasn't working correctly in release build, _probably_ because of R8 despite kotlin-reflect
shipping with Proguard rules and us already keeping all Tachiyomi classes.

arkon committed:
Use Material3 version of AboutLibraries

arkon committed:
Prevent creating backups with no valid options selected

arkon committed:
Try to show actual path in invalid location downloader notification

Instead of the class/hashCode, which doesn't mean much to a user.

arkon committed:
Fix extension settings icon trying to install update instead of opening details

arkon committed:
Defer ACRA reporting until device is idle/not low battery/on unmetered network

arkon committed:
Fix next expected update being weird number sometimes

Occurs if manga.lastUpdate has never been set yet.

arkon committed:
Update Mullvad DoH configuration

Closes #10282

arkon committed:
Remove tmp chapter files after exiting reader

arkon committed:
Force users to retrust unknown extensions on cold starts

arkon committed:
Fix temp chapter files not being able to be created when reading

arkon committed:
Remove custom extension readme/changelog URLs

These were barely used/maintained, so just killing them.
Changelog menu item still exists to take you to the relevant git history.

arkon committed:
Change fetch interval action to show days until next expected update

arkon committed:
Support external repos

Largely taken from SY.

Co-authored-by: jobobby04 <jobobby04@users.noreply.github.com>

arkon committed:
Revert "Implement predictive back animation (#10273)"

This reverts commit 9c120e623193271971448fb03665a73dff4f85cb.

Potentially too buggy for a stable release for now.

arkon committed:
Disable customized fetch intervals for stable builds for now

Until some of the issues get ironed out.

arkon committed:
Fix lint error

arkon committed:
Clean up external repos

- Accept full URL as input instead, which allows for non-GitHub
- Remove automatic CDN fallback in favor of adding that as an external repo if needed

arkon committed:
Bump dependencies

weblate committed:
Translations update from Hosted Weblate (#10244)

Weblate translations



































Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/bn/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ca/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/cs/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/de/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/el/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/es/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/eu/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fil/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/hr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/id/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/it/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ja/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/km/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/nb_NO/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ne/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/pl/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ru/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/sv/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/th/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/tr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/vi/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hans/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hant/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/be/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/es/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/pl/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/ru/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/zh_Hant/
Translation: Tachiyomi/Tachiyomi plurals.xml
Translation: Tachiyomi/Tachiyomi strings.xml

Co-authored-by: Alessandro Jean <14254807+alessandrojean@users.noreply.github.com>
Co-authored-by: Clément de La Bourdonnaye <cle.bourdonnaye@gmail.com>
Co-authored-by: Dexroneum <Rozhenkov69@gmail.com>
Co-authored-by: Dir Sulaiman <sulaiman.tsany@gmail.com>
Co-authored-by: Eduard Ereza Martínez <eduard@ereza.cat>
Co-authored-by: FateXBlood <zecrofelix@gmail.com>
Co-authored-by: Fuxing TAN <tfx1234567@gmail.com>
Co-authored-by: Giorgio Sanna <sannagiorgio1997@gmail.com>
Co-authored-by: Hasanur Rahman Biplob <hrbiplob10@gmail.com>
Co-authored-by: ID-86 <id86dev@gmail.com>
Co-authored-by: InfinityDouki56 <ced.paltep10@gmail.com>
Co-authored-by: ItsPoofy <tuanminh8688@gmail.com>
Co-authored-by: Lyfja <yassinelaoud@gmail.com>
Co-authored-by: Lzmxya <lzmxya@gmail.com>
Co-authored-by: Milo Ivir <mail@milotype.de>
Co-authored-by: Oğuz Ersen <oguz@ersen.moe>
Co-authored-by: Pitpe11 <giorgos2550@gmail.com>
Co-authored-by: Rotakna Oom <oomrotakna11@gmail.com>
Co-authored-by: Sertinel <cankalenderr@yandex.com>
Co-authored-by: Swyter <swyterzone@gmail.com>
Co-authored-by: TheKingTermux <achmadmaulana0233@gmail.com>
Co-authored-by: Unai <uesandi@gmail.com>
Co-authored-by: Uzuki Shimamura <hzy980512@126.com>
Co-authored-by: Vetle Ledaal <vetle.ledaal@gmail.com>
Co-authored-by: altinat <altinat@duck.com>
Co-authored-by: bapeey <90949336+bapeey@users.noreply.github.com>
Co-authored-by: bapeey <luisrleccar@hotmail.com>
Co-authored-by: bittin1ddc447d824349b2 <bittin@reimu.nl>
Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: inson1 <vaclav.svarc01@seznam.cz>
Co-authored-by: orkan gökçe alaz aşina <examplehuman@outlook.com>
Co-authored-by: sebastians17 <sebastians117.ss@gmail.com>
Co-authored-by: Макар Разин <makarrazin14@gmail.com>

arkon committed:
Bump minimum WebView version and default user agent string

arkon committed:
Release v0.15.0

arkon committed:
Reword onboarding prompt for returning users

arkon committed:
Add link to storage guide during onboarding step

arkon committed:
Avoid showing WebView button in reader errors if page isn't actually from web

arkon committed:
Fix RAR loading

Closes #10302

arkon committed:
Show non-localized language names too in app language selection

arkon committed:
Normalize some locale names

arkon committed:
Avoid hard crash if cached image file was already deleted

Closes #9720

arkon committed:
Add link to storage FAQ in settings screen

arkon committed:
Allow opening .tachibk files directly with app to restore

weblate committed:
Translations update from Hosted Weblate (#10294)

Weblate translations















Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/es/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fil/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/id/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/nb_NO/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/pt/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ru/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hans/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hant/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/fr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/id/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/ko/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/zh_Hans/
Translation: Tachiyomi/Tachiyomi plurals.xml
Translation: Tachiyomi/Tachiyomi strings.xml

Co-authored-by: Alessandro Jean <14254807+alessandrojean@users.noreply.github.com>
Co-authored-by: Dexroneum <Rozhenkov69@gmail.com>
Co-authored-by: InfinityDouki56 <ced.paltep10@gmail.com>
Co-authored-by: Luigi <luigi.joubert@gmail.com>
Co-authored-by: Lzmxya <lzmxya@gmail.com>
Co-authored-by: Matheus Victor Ramos dos Anjos <matheusvra@hotmail.com>
Co-authored-by: Swyter <swyterzone@gmail.com>
Co-authored-by: Vetle Ledaal <vetle.ledaal@gmail.com>
Co-authored-by: Yefita <Yefita@users.noreply.hosted.weblate.org>
Co-authored-by: Zero O <godarms2010@live.com>
Co-authored-by: bapeey <90949336+bapeey@users.noreply.github.com>
Co-authored-by: moon <moononclouds@proton.me>
Co-authored-by: stevenlele <stevenlele@outlook.com>

arkon committed:
Release v0.15.1

arkon committed:
Allow deep linking to add external repo

arkon committed:
Avoid jank in extensions list if install permissions already granted

arkon committed:
Allow permanently trusting unofficial extensions by version code + signature

Closes #10290

Riztard committed:
Add Nord Theme (#10308)

* Add Nord Theme

* update

* update error color

* update comment

arkon committed:
Fix selecting custom fetch interval not persisting sometimes

arkon committed:
Add filter library by customized update frequency

Supersedes #9619

Co-authored-by: quangkieu <quangkieu@users.noreply.github.com>

arkon committed:
Trusting new extension shouldn't revoke other irrelevant extensions

arkon committed:
Add advanced setting to revoke all trusted unknown extensions

arkon committed:
Fix external repo info banner in ExtensionDetailsScreen

arkon committed:
Do proper check for next chapter's download status when downloading ahead

Fixes #10151 (I think?)

arkon committed:
Avoid floating point precision issues when converting scores

Fixes #10343
Maybe we'll finally migrate off of those legacy models some day...

renovate[bot] committed:
Update dependency io.mockk:mockk to v1.13.9 (#10349)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

arkon committed:
Remove built-in official extension repo support

CalebMorris committed:
[track-search] Added context menu for copy and open-in-web (#10352)

Naputt1 committed:
fix Can't scroll down to the bottom of the webtoon on the last chapter (#10291)

arkon committed:
Move extension repos interactors to proper package

Also retain ordering of added repos.

arkon committed:
[skip ci] update issue templates

arkon committed:
Disable some non-ready stuff for stable build

weblate committed:
Translations update from Hosted Weblate (#10336)

Weblate translations
















Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ca/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/de/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/eo/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/es/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fil/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/nb_NO/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ru/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/th/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/tr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hans/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hant/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/ca/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/nb_NO/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/th/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/tr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/zh_Hans/
Translation: Tachiyomi/Tachiyomi plurals.xml
Translation: Tachiyomi/Tachiyomi strings.xml

Co-authored-by: Alessandro Jean <14254807+alessandrojean@users.noreply.github.com>
Co-authored-by: Dexroneum <Rozhenkov69@gmail.com>
Co-authored-by: Eduard Ereza Martínez <eduard@ereza.cat>
Co-authored-by: InfinityDouki56 <ced.paltep10@gmail.com>
Co-authored-by: Lyfja <yassinelaoud@gmail.com>
Co-authored-by: Lzmxya <lzmxya@gmail.com>
Co-authored-by: Oğuz Ersen <oguz@ersen.moe>
Co-authored-by: Radoŝ Porka <animatorzPolski@gmail.com>
Co-authored-by: Swyter <swyterzone@gmail.com>
Co-authored-by: Vetle Ledaal <vetle.ledaal@gmail.com>
Co-authored-by: Zero O <godarms2010@live.com>
Co-authored-by: altinat <altinat@duck.com>
Co-authored-by: bapeey <90949336+bapeey@users.noreply.github.com>
Co-authored-by: gallegonovato <fran-carro@hotmail.es>

arkon committed:
Release v0.15.2

arkon committed:
Minor extension repo enhancements

- Shortcut to settings from extensions tab
- Don't show error toast anymore if nothing's loaded
- Ability to copy extension repo URL to clipboard

CalebMorris committed:
Fixed dev UI preview (#10385)

The TachiyomiTheme introduced a dependency-injection construct that didn't
 exist at the time of rendering previews, so I've changed the preview function
 to use a preview version of the theme that uses declarative configuration
 over dependency injection

stevenyomi committed:
Fix extension interceptors receiving compressed responses (#10388)

kaiserbh committed:
Use transaction on restore to go brr. (#10375)

refactor: use transaction to go brr.

This improve the restore speed on fresh db and non fresh db.

Signed-off-by: KaiserBh <kaiserbh@proton.me>

CalebMorris committed:
[download-cache] Fixed init logic to skip when cache file is missing (#10362)

There are several possible causes of the cache file to not exist, including user
 action. By skipping these couple steps during initialization when the file is
 missing, a renew action is allowed to start and the cache will rebuild and
 hopefully work as expected.

Simple fix for #10360

weblate committed:
Translations update from Hosted Weblate (#10386)

Weblate translations































Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/cs/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/de/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/el/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/eo/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/es/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fil/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/hr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/id/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ja/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ko/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ms/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ne/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/pt/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/pt_BR/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ru/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/sv/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/te/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/th/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/tr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hans/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/zh_Hant/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/cs/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/de/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/el/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/hr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/ja/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/ms/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/ne/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/pt/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/zh_Hans/
Translation: Tachiyomi/Tachiyomi plurals.xml
Translation: Tachiyomi/Tachiyomi strings.xml

Co-authored-by: Alessandro Jean <14254807+alessandrojean@users.noreply.github.com>
Co-authored-by: Andreas E <andreas.everos@gmail.com>
Co-authored-by: CodeSpoof <nao.s_l_t_e_e_l@protonmail.com>
Co-authored-by: Crazyom <naxom@laposte.net>
Co-authored-by: Daedren <lord.raikon@gmail.com>
Co-authored-by: DarKCroX <DarKCroX@users.noreply.hosted.weblate.org>
Co-authored-by: Dexroneum <Rozhenkov69@gmail.com>
Co-authored-by: Gianluca Starke <gianlucastarke@gmail.com>
Co-authored-by: ID-86 <id86dev@gmail.com>
Co-authored-by: InfinityDouki56 <ced.paltep10@gmail.com>
Co-authored-by: Khori Hutama <khori.qq@gmail.com>
Co-authored-by: Kirito ._ <kiritokunn18@gmail.com>
Co-authored-by: Lyfja <yassinelaoud@gmail.com>
Co-authored-by: Lzmxya <lzmxya@gmail.com>
Co-authored-by: Manjul Tamrakar <manjultamrakar4@gmail.com>
Co-authored-by: Milo Ivir <mail@milotype.de>
Co-authored-by: Oğuz Ersen <oguz@ersen.moe>
Co-authored-by: Paavalen Lingachetti <p.lingachetti@gmail.com>
Co-authored-by: Pitpe11 <giorgos2550@gmail.com>
Co-authored-by: Radoŝ Porka <animatorzPolski@gmail.com>
Co-authored-by: Sup Kelelawar <apkfile007@gmail.com>
Co-authored-by: Swyter <swyterzone@gmail.com>
Co-authored-by: Uzuki Shimamura <hzy980512@126.com>
Co-authored-by: Zero O <godarms2010@live.com>
Co-authored-by: altinat <altinat@duck.com>
Co-authored-by: dan-malprod <diabolic0240@proton.me>
Co-authored-by: jinu147 <nesqea20@gmail.com>
Co-authored-by: orkan gökçe alaz aşina <examplehuman@outlook.com>
Co-authored-by: stevenlele <stevenlele@outlook.com>

arkon committed:
Fixed tap controls not working when zoomed in (#10378)

Co-authored-by: Paloys <Paloys@users.noreply.github.com>

arkon committed:
Clean up some unnecessary bits

- Remove analytics/crash reporting
- Remove app update check
- Remove F-Droid warning
- Remove Discord references

arkon committed:
Update Compose compiler and Kotlin

arkon committed:
Better message for empty extensions list

weblate committed:
Translations update from Hosted Weblate (#10393)

Weblate translations












Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/bg/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ca/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/de/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/fil/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/hr/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/id/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/it/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/ru/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/strings/uk/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/bg/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/de/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/eo/
Translate-URL: https://hosted.weblate.org/projects/tachiyomi/tachiyomi-plurals-xml/fil/
Translation: Tachiyomi/Tachiyomi plurals.xml
Translation: Tachiyomi/Tachiyomi strings.xml

Co-authored-by: Boyan Alexiev <nneauu@gmail.com>
Co-authored-by: Denis \"Samilton <d.bogdan99@gmail.com>
Co-authored-by: Dexroneum <Rozhenkov69@gmail.com>
Co-authored-by: Eduard Ereza Martínez <eduard@ereza.cat>
Co-authored-by: Giorgio Sanna <sannagiorgio1997@gmail.com>
Co-authored-by: Lyfja <yassinelaoud@gmail.com>
Co-authored-by: Milo Ivir <mail@milotype.de>
Co-authored-by: Radoŝ Porka <animatorzPolski@gmail.com>
Co-authored-by: Shiratori <kuromaruhatake@gmail.com>
Co-authored-by: TheKingTermux <achmadmaulana0233@gmail.com>

arkon committed:
Release v0.15.3

arkon committed:
Use newer build tools

arkon committed:
Install build-tools 29.0.3
